### PR TITLE
CMake: improve handling of options

### PR DESCRIPTION
**Pull request by mic84**
* Refactor AMReX_Options.cmake
* Better handling of dependent options via cmake_dependent_option()
* Improve interaction between xSDK and non-xSDK options.

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -74,6 +74,18 @@ if ( CMAKE_SOURCE_DIR STREQUAL PROJECT_SOURCE_DIR AND CMAKE_INSTALL_PREFIX_INITI
...
 endif ()
 
+message(STATUS "AMReX installation directory: ${CMAKE_INSTALL_PREFIX}")
+
+#
+# Check if CMAKE_BUILD_TYPE is given. If not, use default
+#
+if ( NOT CMAKE_BUILD_TYPE )
+   message(STATUS "Setting build type to Release as none was specified.")
+   set( CMAKE_BUILD_TYPE Release )
```
Comment (ax3l) [inline CMakeLists.txt]: I usually do this without a message just to change the default and keep the options alive as cache var:
```suggestion
   set(CMAKE_CONFIGURATION_TYPES "Release;Debug;MinSizeRel;RelWithDebInfo")
   set(CMAKE_BUILD_TYPE Release
       CACHE STRING
       "Choose the build type, e.g. Release, Debug, or RelWithDebInfo." FORCE)
```

File: Tools/CMake/AMReX_Options.cmake [hunk 2]
```diff
@@ -39,12 +24,21 @@ message(STATUS "Configuring AMReX with the following options: ")
 #
 # This is the option to enable/disable xSDK mode
 #
+# To handle both XDSK options and correponding plain AMReX options,
+# we make use of policy CMP0077 introduced as a default in CMake 3.13
+# Under policy CMP0077, normal variables prevent option()
+# to set internal variables of the same name.
+# Example: if XSDK mode is ON and XSDK_ENABLE_Fortran=ON, a normal
+# variable ENABLE_FORTRAN will be created and se to ON.
```
Comment (ax3l) [inline Tools/CMake/AMReX_Options.cmake]:
```suggestion
# variable ENABLE_FORTRAN will be created and set to ON.
```

File: Tools/CMake/AMReX_Options.cmake [hunk 3]
```diff
@@ -192,81 +188,45 @@ print_option( ENABLE_ASSERTIONS )
...
 option( ENABLE_BACKTRACE "Enable backtracing" OFF)
 print_option( ENABLE_BACKTRACE )
 
-option( ENABLE_PROFPARSER "Enable profile parser" OFF)
+cmake_dependent_option(ENABLE_PROFPARSER "Enable profile parser" OFF
+   "ENABLE_BASE_PROFILE;ENABLE_TRACE_PROFILE;ENABLE_AMRDATA" OFF)
 print_option( ENABLE_PROFPARSER )
 
-set ( TP_PROFILE "None" CACHE STRING "Third-party profiling options:<CRAYPAT,FORGE,VTUNE>")
+set( TP_PROFILE "None" CACHE STRING "Third-party profiling options:<CRAYPAT,FORGE,VTUNE>")
```
Comment (ax3l) [inline Tools/CMake/AMReX_Options.cmake]: Straight from Craig Scott https://stackoverflow.com/a/39671055/2719194 we can set a GUI helper and validate against the values below with a list:
```suggestion
set(TP_PROFILE_VALUES None CRAYPAT FORGE VTUNE)
set(TP_PROFILE None CACHE STRING "Third-party profiling options: <CRAYPAT,FORGE,VTUNE>")
set_property(CACHE TP_PROFILE PROPERTY STRINGS ${TP_PROFILE_VALUES})
if(NOT TP_PROFILE IN_LIST TP_PROFILE_VALUES)
    message(FATAL_ERROR "TP_PROFILE (${TP_PROFILE}) must be one of ${TP_PROFILE_VALUES}")
endif()
```

Comment (mic84) [reply Tools/CMake/AMReX_Options.cmake]: Looks good to me! I would keep the available options, i.e. "Third-party profiling options:<CRAYPAT,FORGE,VTUNE>", in the doc string for cache variable TP_PROFILE.

Comment (ax3l) [reply Tools/CMake/AMReX_Options.cmake]: Oh right, I though `ccmake` shows them but maybe that's just a GUI thingy or no feature at all.

**Pull request merged**